lib/autocomplete.ts

```diff
@@ -90,8 +99,9 @@ class Autocomplete extends BaseMenu implements IPopupContent {
     this.autoDispose(onKeyElem(trigger, 'keydown', {
       ArrowDown: () => this.nextIndex(),
       ArrowUp: () => this.prevIndex(),
-      Enter$: () => this._selected && this._selected.click(),
-      Escape: () => ctl.close(0)
+      // On Enter key we only update the value and let the event propagate for the consumer to
+      // handle it directly.
+      Enter$: () => this._selected && (trigger.value = dom.getData(this._selected, 'choice')),
```

Comment:
Is this handler even needed at all given that by default Enter causes the `menuItem`'s action to be run, which does what we want?

Reply:
`menuItem`'s action do not run on `Enter` because the focus remains on the input element. I've added a comment.